services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/manage_units/unit.tsx

```diff
@@ -0,0 +1,229 @@
+    return cuId || classroom_unit_id
+  }
+
+  function renderModal() {
+    const activities = getClassroomActivities()
+    const lessonActivityData = activities.get(lessonActivityId)
+
+    if (!showModal || !(Object.keys(lessonActivityData).length)) { return }
+
+    return (
+      <PreviewOrLaunchModal
+        classroomUnitId={getClassroomUnitId(lessonActivityData)}
+        closeModal={handleToggleModal}
+        completed={lessonActivityData.completed}
+        lessonID={lessonActivityId}
+      />
+    );
+  }
+
+  function renderActivityCell(lessonActivity, id) {
+    const { name, completed, started } = lessonActivity
+    const href = `/teachers/classroom_units/${getClassroomUnitId(lessonActivity)}/mark_lesson_as_completed/${id}`;
+    return(
+      <div className="activity-cell-container">
+        <DataTableChip color="green" label={name} onClick={(e) => handleDataTableChipClick(e, id)} />
+        {completed && <div className="lesson-complete"><p>Lesson Complete</p><img alt="" src={`${EXTRA_SMALL_ICON_BASE_SRC}/check-circle.svg`} /></div>}
+        {started && <a className={sharedButtonClassName} href={href} rel="noopener noreferrer" target="_blank">Mark As Complete</a>}
+      </div>
+    )
+  }
+
+  function handleLaunchLessonWithNoStudents(e, activityId, classroomUnitId) {
+    if (window.confirm("You have no students in this class. Quill Lessons is a collaborative tool for teachers and students to work together. If you'd like to launch this lesson anyway, click OK below. Otherwise, click Cancel.")) {
+      window.location.href = `${process.env.DEFAULT_URL}/teachers/classroom_units/${classroomUnitId}/launch_lesson/${activityId}`;
+    }
+  }
+
+  function renderLaunchLessonCell(lessonActivity) {
+    const { completed, studentCount, started } = lessonActivity
+    const activityId = getActivityId(lessonActivity)
+    const classroomUnitId = getClassroomUnitId(lessonActivity)
+    if (completed) {
+      return <a className={sharedButtonClassName} href={`/teachers/progress_reports/report_from_classroom_unit_and_activity/${classroomUnitId}/a/${activityId}`} rel="noopener noreferrer" target="_blank">View Report</a>;
+    }
+    if (studentCount === 0) {
+      return <a className={sharedButtonClassName} id="launch-lesson" onClick={(e) => handleLaunchLessonWithNoStudents(e, activityId, classroomUnitId)}>{started ? 'Resume Lesson' : 'Launch Lesson'}</a>;
+    }
+    if (started) {
+      return <a className={sharedButtonClassName} href={`${process.env.DEFAULT_URL}/teachers/classroom_units/${classroomUnitId}/launch_lesson/${activityId}`}>Resume Lesson</a>;
+    }
+    return <a className={sharedButtonClassName} href={`${process.env.DEFAULT_URL}/teachers/classroom_units/${classroomUnitId}/launch_lesson/${activityId}`}>Launch Lesson</a>;
+  }
+
+  function renderCustomizeCell(id) {
+    return(
+      <a className="customize-link focus-on-light" href={`/customize/${id}`} rel="noopener noreferrer" target="_blank">
```

Comment:
We should add an `aria-label` to these tags, because it won't be clear to a screenreader user what clicking that icon will do. Then, I'd test for the presence of an element with that aria label in the specs, instead of the alt text, which is less relevant for functionality.

Reply:
I'll add, thanks for the context!